massa-protocol-worker/src/protocol_network.rs

```diff
@@ -242,7 +242,15 @@ impl ProtocolWorker {
     ) -> Result<(), ProtocolError> {
         if let Some(info) = self.block_wishlist.get(&block_id) {
             if info.header.is_some() {
-                let _ = self.ban_node(&from_node_id).await;
+                warn!(
+                    "Node {} sent us header for block id {} but we already received it.",
+                    from_node_id, block_id
+                );
+                if let Some(node) = self.active_nodes.get_mut(&from_node_id) && node.asked_blocks.contains_key(&block_id) {
+                    node.asked_blocks.remove(&block_id);
+                    node.insert_known_blocks(&[block_id], false, Instant::now(), self.config.max_node_known_blocks_size);
+                }
+
```

Comment:
shall we run the ask_for_block here ?

Reply:
It's running automatically at short intervals and if received need connection/block header etc so I think we don't need to add a run here

Reply:
okay